tests/test_custom_middleware_exception.py

```diff
@@ -56,6 +56,12 @@ async def __call__(self, scope, receive, send):
 
 @router.post("/middleware")
 def run_middleware(file: UploadFile = File(..., description="Big File")):
+    # TODO: without this, pytest raises an error, and it causes a segmentation fault on
+    # Python 3.11-rc.2, the app works normally when run with Uvicorn, not sure if it's
+    # a problem with Python 3.11, pytest, or FastAPI.
+    # But FastAPI (or Starlette) should probably try to close these
+    # SpooledTemporaryFiles automatically by default
+    file.file.close()
```

Comment:
I think it's on starlette

Reply:
This was handled here: https://github.com/tiangolo/fastapi/pull/5465 :tada: